containerd/driver.go

```diff
@@ -151,23 +153,24 @@ type Mount struct {
 // TaskConfig contains configuration information for a task that runs with
 // this plugin
 type TaskConfig struct {
-	Image          string   `codec:"image"`
-	Command        string   `codec:"command"`
-	Args           []string `codec:"args"`
-	CapAdd         []string `codec:"cap_add"`
-	CapDrop        []string `codec:"cap_drop"`
-	Cwd            string   `codec:"cwd"`
-	Devices        []string `codec:"devices"`
-	Seccomp        bool     `codec:"seccomp"`
-	SeccompProfile string   `codec:"seccomp_profile"`
-	Privileged     bool     `codec:"privileged"`
-	PidsLimit      int64    `codec:"pids_limit"`
-	HostDNS        bool     `codec:"host_dns"`
-	ExtraHosts     []string `codec:"extra_hosts"`
-	Entrypoint     []string `codec:"entrypoint"`
-	ReadOnlyRootfs bool     `codec:"readonly_rootfs"`
-	HostNetwork    bool     `codec:"host_network"`
-	Mounts         []Mount  `codec:"mounts"`
+	Image          string             `codec:"image"`
+	Command        string             `codec:"command"`
+	Args           []string           `codec:"args"`
+	CapAdd         []string           `codec:"cap_add"`
+	CapDrop        []string           `codec:"cap_drop"`
+	Cwd            string             `codec:"cwd"`
+	Devices        []string           `codec:"devices"`
+	Seccomp        bool               `codec:"seccomp"`
+	SeccompProfile string             `codec:"seccomp_profile"`
+	Sysctl         hclutils.MapStrStr `codec:"sysctl"`
```

Comment:
Why is this a `MapStrStr` while the above hclspec has `list(map(string))` ?

Reply:
This is how it is in `docker driver`

https://github.com/hashicorp/nomad/blob/main/drivers/docker/config.go#L396 and https://github.com/hashicorp/nomad/blob/main/drivers/docker/config.go#L465

This works pretty well, as the input (`sysctl`) is directly populated into a `go` map, which I can use to set the `map` in the container spec.